client/src/components/App.js

```diff
@@ -14,16 +15,33 @@ import Dashboard from './Dashboard/Dashboard';
 import Personal from './Personal/Personal';
 
 class App extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            currentURL: "/"
+        }
+    }
     componentDidMount() {
         this.props.fetchUser();
+
+        // Listen to the URL for route changes and update the currentURL state accordingly
+        this.unlisten = history.listen((location) => {
+            this.setState({ currentURL: location.pathname });
+        });
```

Comment:
Our root component will listen to the History object and if there's a route change, we'll set the "currentURL"

We'll then hide the Header.js ONLY if we're in the error's page (for styling purposes...Don't want that navigation bar in the error page)